src/inventory.cpp

```diff
@@ -50,40 +50,46 @@ static std::string nameFromPath(const std::string& path)
  */
 static bool humanCompare(const InventoryItem& a, const InventoryItem& b)
 {
-    static const char* digits = "0123456789";
-
-    size_t lastPos = 0;
-    while (lastPos < a.name.length())
+    const char* strA = a.name.c_str();
+    const char* strB = b.name.c_str();
+    while (true)
     {
-        // searching for start of digits inside the name
-        const size_t start = a.name.find_first_of(digits, lastPos);
-        if (start == std::string::npos ||
-            start != b.name.find_first_of(digits, lastPos))
+        const char& chrA = *strA;
+        const char& chrB = *strB;
+
+        // check for end of name
+        if (!chrA || !chrB)
+            return chrB;
+
+        const bool isNumA = (chrA >= '0' && chrA <= '9');
+        const bool isNumB = (chrB >= '0' && chrB <= '9');
+
+        if (isNumA & isNumB)
         {
-            break;
+            // both names have numbers at the same position
+            char* endA = nullptr;
+            char* endB = nullptr;
+            const unsigned long valA = strtoul(strA, &endA, 10);
+            const unsigned long valB = strtoul(strB, &endB, 10);
+            if (valA != valB)
+                return valA < valB;
+            strA = endA;
+            strB = endB;
         }
-
-        // check for equalty of names before digits
-        const size_t partSize = start - lastPos;
-        if (a.name.compare(lastPos, partSize, b.name, lastPos, partSize) != 0)
+        else if (isNumA ^ isNumB)
         {
-            break;
+            // only one of names has a number
+            return isNumB;
```

Comment:
Here should be `return isNumA;` or it gives wrong order. 
Example:
```
 PSU0_VIN
 PSU0_VOUT
 PSU1_VIN
 PSU1_VOUT
 P0V89_CPU0
 P0V89_CPU1
 P0V89_CPU2
 P0V89_CPU3
 P0V97_CPU0
 P0V97_CPU1
 P0V97_CPU2
 P0V97_CPU3
```

Reply:
Done